doc/user/layouts/partials/create-source/connector/kafka/with-options.html

```diff
@@ -4,6 +4,7 @@
 `security_protocol` | `text` | Use [`ssl`](#ssl-with-options) or, for [Kerberos](#kerberized-kafka-details), `sasl_plaintext`, `sasl-scram-sha-256`, or `sasl-sha-512` to connect to the Kafka cluster.
 `statistics_interval_ms` | `int` | `librdkafka` statistics emit interval in `ms`. Accepts values [0, 86400000]. The granularity is 1000ms. A value of 0 disables statistics. Statistics can be queried using the `mz_kafka_consumer_partitions` system table.
 `ignore_source_keys` | `boolean` | Default: `false`. If `true`, do not perform optimizations assuming uniqueness of primary keys in schemas.
+`start_offset`| `int` | Read partitions from the specified offset. If only one partition is specified, the order reverts to latest-first after the specified partition. If a topic currently has only one partition, the offset will apply to a future partition. You cannot update the offsets once a source has been created; you will need to recreate the source. Values must be zero or positive integers.
```

Comment:
maybe I'm missing something but "If only one partition is specified, the order reverts to latest-first after the specified partition." and "If a topic currently has only one partition, the offset will apply to a future partition." seem a little confusing next to each other

I'm not entirely sure what the first sentence means -- specifically "order reverts to latest-first" is tripping me up. We never read messages in a latest first order, we always read them earliest first, but sometimes with an offset. More concretely if a kafka topic has messages 

a, b, c, d, e at offsets 0, 1, 2, 3, 4 respectively, then if we started reading at offset 2, we are guaranteed to read c, d, e in that order.

"If a topic currently has only one partition, the offset will apply to a future partition." I'm not sure if that's true but if so, it seems like a confusing thing / misfeature. That implies that if we specify a start offset for exactly one partition, then that offset applies for all future partitions in the topic, but then if we specify multiply multiple start offsets for multiple partitions I'm not sure what happens. Let me double check the code to see what we actually do there

Reply:
I removed "If only one partition is specified, the order reverts to latest-first after the specified partition" (that one is my error),  but "If a topic currently has only one partition, the offset will apply to a future partition" was a gotcha @JLDLaughlin specifically warned me about. I may have misunderstood what she was saying, though.

Reply:
I double checked the code and I don't think we do that behavior anymore. If a start offset is not specified for a partition, we will read it from the beginning. (I think previously we used to apply the start offset to other partitions but we don't anymore)